### PR TITLE
Add support for minor ticks in 3d axes.

### PR DESCRIPTION
**Pull request by anntzer**
The actual implementation is shorter than deleting the copies of "Minor
ticks are not supported" in the various docstrings...
Edit: Just a bit longer now that minor ticks are style differently.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by WeatherGod** [comment]
I am surprised that it is this simple to get the feature roughly working. Maybe the api has gotten better at this than when I last looked at this. I still have a vague recollection that I encountered a deeper problem when I looked at this 8 years ago.

I do want to try and get this to look right, though. Right now, the minor/major ticks can't be distinguished, and the text labels all overlap each other. The code for handling tick marks and labels all assume major tick semantics. We might need to make a few more changes to get this to look "right".

**Comment by anntzer** [comment]
There isn't much to be done about overlapping text (you'd get the same with long labels in 2D).  I can probably switch minor ticks to use a length of xtick.minor.size and a linewidth of xtick.minor.width, but that's all the low lying fruits I can think of.

**Comment by anntzer** [comment]
@ImportanceOfBeingErnest you :+1:'d this, do you want to review it? :) or @WeatherGod?

File: lib/mpl_toolkits/tests/test_mplot3d.py [hunk 1]
```diff
@@ -986,3 +986,14 @@ def test_quiver3D_smoke(fig_test, fig_ref):
     for fig, length in zip((fig_ref, fig_test), (1, 1.0)):
         ax = fig.gca(projection="3d")
         ax.quiver(x, y, z, u, v, w, length=length, pivot=pivot)
+
+
+@image_comparison(["minor_ticks.png"], style="default")
```
Comment (QuLogic) [inline lib/mpl_toolkits/tests/test_mplot3d.py]: I think we want to use `mpl20` now, don't we?

Comment (anntzer) [reply lib/mpl_toolkits/tests/test_mplot3d.py]: either way works for me.

**Comment by anntzer** [comment]
rebased

**Comment by QuLogic** [comment]
Seems to have really broken tests.

**Comment by anntzer** [comment]
likely because this crossed wires with the 3d projection change -- regenerated the baseline.

**Pull request merged**